content/4_Gold/Tree_Euler.mdx

```diff
@@ -715,6 +715,81 @@ preprocessing and query in $\mathcal{O}(1)$ time.
 
 <FocusProblem problem="rmqSample" />
 
+The following is an example implementation of a sparse table and code that
+answers LCA queries. Build time is $\mathcal{O}(N\log N)$, and queries are
+$\mathcal{O}(1)$.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+template <class T> struct RMQ {
+	int n, log2dist;
+	vector<vector<T>> st;
+	/** constructor for the class */
+
+	RMQ(const vector<T> &v) {
+		n = (int)v.size();
+		log2dist = 1 + (int)log2(n);
```

Comment:
just ceil is fine idk

Reply:
nah, i ran it on yosupo and setting log2dist to (int) ceil(log2(n)) causes RTE

Reply:
LOL WHAT

Reply:
wait no i’m stupid i think i submitted my lca code to static rmq LMFAO

Reply:
i’ll change it later 💀

Reply:
https://judge.yosupo.jp/submission/211411 i lied lmao my filter settings were wrong, it does in fact rte

Reply:
rte?

Reply:
yeah, idk why it does